controllers/taloscontrolplane_controller.go

```diff
@@ -293,6 +294,16 @@ func (r *TalosControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.Re
 
 		return res, err
 	default:
+		if conditions.IsFalse(tcp, controlplanev1.MachinesBootstrapped) {
```

Comment:
my concern here (using conditions to track bootstrap status) is that conditions (as whole status) doesn't get preserved when CAPI is pivoted to another cluster

So it might issue `bootstrapCluster` once again, Talos has handling for it (it will refuse to bootstrap if etcd is running), but we should handle this error and mark condition as bootstrapped in that case.